lib/Newsletter/Sending/ScheduledTasksRepository.php

```diff
@@ -87,6 +89,45 @@ public function findPreviousTask(ScheduledTaskEntity $task): ?ScheduledTaskEntit
       ->getOneOrNullResult();
   }
 
+  public function findDueByType($type, $limit = null) {
+    return $this->findByTypeAndStatus($type, ScheduledTaskEntity::STATUS_SCHEDULED, $limit);
+  }
+
+  public function findRunningByType($type, $limit = null) {
+    return $this->findByTypeAndStatus($type, null, $limit);
+  }
+
+  protected function findByTypeAndStatus($type, $status, $limit = null, $future = false) {
+    $queryBuilder = $this->doctrineRepository->createQueryBuilder('st')
+      ->select('st')
+      ->where('st.type = :type')
+      ->setParameter('type', $type)
+      ->andWhere('st.deletedAt IS NULL');
+
+    if (is_null($status)) {
+      $queryBuilder->andWhere('st.status IS NULL');
+    } else {
+      $queryBuilder
+        ->andWhere('st.status = :status')
+        ->setParameter('status', $status);
+    }
+
+    if ($future) {
+      $queryBuilder->andWhere('st.scheduledAt >= :now');
```

Comment:
I noticed a small difference when comparing with [the old model code](https://github.com/mailpoet/mailpoet/blob/master/lib/Models/ScheduledTask.php#L189). In the old model, we use `whereGt` which is an equivalent for `>`. 

This is just a small difference but theoretically, it may change how the plugin triggers cron for jobs that are scheduled immediately using the current time. With `>` a significant portion of such jobs might be skipped in the current HTTP request (the one they were scheduled in) and triggered in a subsequent one.

Maybe I'm too much paranoid, but I would keep the behavior as close as possible to the current state. What do you think?

Reply:
Oh! I've just realized that it is the other way around :) Sorry! We trigger jobs cron for jobs that are due so this has nothing to do with the future jobs. But still, it might be better to keep those future and non-future intervals without an intersection.

Reply:
Good catch! I missed this when refactoring the code. I just pushed a new commit to fix this problem. Could you please review the PR again?